runtime/runtime-params-estimator/src/replay/fold_db_ops.rs

```diff
@@ -0,0 +1,263 @@
+use super::cache_stats::CacheStats;
+use super::Visitor;
+use std::collections::BTreeMap;
+use std::io::Write;
+
+const EMPTY_STATE: &str = "states must never be empty";
+
+/// A visitor that keeps track of DB operations and aggregates it by specific labels.
+pub(super) struct FoldDbOps {
+    /// Labels at which to break aggregation.
+    fold_anchors: Vec<String>,
+    /// Fields to display on each output, if they are available in the trace.
+    printed_fields: Vec<String>,
```

Comment:
Can we store it as a mapping from anchor to printed fields?

I see that it doesn't impact our usecases, but I think it makes code more readable. Initially I got confused why do we have a duplicated printed field `fn receipts(self)`, and printed fields are connected to specific anchor anyway.

Reply:
yes that make sense, I didn't even notice the duplicate output, that was a bug too^^
anyway, using a map makes more sense